pkg/resources/secret_with_oauth_authorization_code_grant_acceptance_test.go

```diff
@@ -187,3 +239,77 @@ func TestAcc_SecretWithAuthorizationCodeGrant_DifferentTimeFormats(t *testing.T)
 		},
 	})
 }
+
+func TestAcc_SecretWithAuthorizationCodeGrant_ExternalChange(t *testing.T) {
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	name := id.Name()
+	comment := random.Comment()
+	refreshTokenExpiryDateTime := time.Now().Add(24 * time.Hour).Format(time.DateTime)
+	newRefreshTokenExpiryDateOnly := time.Now().Add(4 * 24 * time.Hour).Format(time.DateOnly)
+	refreshToken := "test_token"
+
+	integrationId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	_, apiIntegrationCleanup := acc.TestClient().SecurityIntegration.CreateApiAuthenticationClientCredentialsWithRequest(t,
+		sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(integrationId, true, "foo", "foo"),
+	)
+	t.Cleanup(apiIntegrationCleanup)
+
+	secretModel := model.SecretWithAuthorizationCodeGrant("s", integrationId.Name(), id.DatabaseName(), id.SchemaName(), name, refreshToken, refreshTokenExpiryDateTime).WithComment(comment)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.SecretWithAuthorizationCodeGrant),
+		Steps: []resource.TestStep{
+			// create
+			{
+				Config: config.FromModel(t, secretModel),
+				Check: resource.ComposeTestCheckFunc(
+					assert.AssertThat(t,
+						resourceassert.SecretWithAuthorizationCodeResource(t, secretModel.ResourceReference()).
+							HasNameString(name).
+							HasDatabaseString(id.DatabaseName()).
+							HasSchemaString(id.SchemaName()).
+							HasApiAuthenticationString(integrationId.Name()).
+							HasOauthRefreshTokenString(refreshToken).
+							HasOauthRefreshTokenExpiryTimeString(refreshTokenExpiryDateTime).
+							HasCommentString(comment),
+
+						resourceshowoutputassert.SecretShowOutput(t, secretModel.ResourceReference()).
+							HasName(name).
+							HasDatabaseName(id.DatabaseName()).
+							HasSecretType("OAUTH2").
+							HasSchemaName(id.SchemaName()).
+							HasComment(comment),
+					),
+				),
+			},
+			{
+				PreConfig: func() {
+					acc.TestClient().Secret.Alter(t, sdk.NewAlterSecretRequest(id).WithSet(*sdk.NewSecretSetRequest().
```

Comment:
This test is not entirely accurate. Here we change BOTH config and object state in Snowflake, so we would have the value from the new config anyway (external change is not taken into account). We separate two cases: one for changing only the config and checking it as usual, and one for changing it only externally and checking, if we set back to the old value.

Reply:
You're right, changed the test so it checks solely for the external changes, not both config and external